Breaking up envvars and config settings into separate settings pages (#1028)
minio/docs: source/reference/minio-server/settings/metrics-and-logging.rst

.. _minio-server-envvar-metrics-logging:

======================================================

> Environment Variables to Configure Metrics and Logging

Reviewer: We have configuration settings on this page too.

"Metrics and Logging Settings"

> ======================================================

Reviewer: We need to work on the page structure here.

The H1s don't 1:1 nicely - `Environment Variables -> Server Logs` when the config analog is `Configuration Values - HTTP Webhook Log Target`.

This is another area where I'd recommend swapping the H1s and H2s:

```
# Prometheus Authentication

# Server Webhook Logging

## Envvar

## Config Setting

# Audit Webhook Logging

## Envvar

## Config Setting

```

   This environment variable corresponds with the :mc-conf:`audit_kafka.version` configuration setting.

.. envvar:: MINIO_AUDIT_KAFKA_COMMENT
   :optional:

   .. include:: /includes/common-mc-admin-config.rst
      :start-after: start-minio-kafka-audit-logging-comment-desc
      :end-before: end-minio-kafka-audit-logging-comment-desc

   This environment variable corresponds with the :mc-conf:`audit_kafka.comment` configuration setting.

.. envvar:: MINIO_AUDIT_KAFKA_QUEUE_DIR
   :optional:

   .. include:: /includes/common-mc-admin-config.rst
      :start-after: start-minio-kafka-audit-logging-queue-dir-desc
      :end-before: end-minio-kafka-audit-logging-queue-dir-desc

   This environment variable corresponds with the :mc-conf:`audit_kafka.queue_dir` configuration setting.

.. envvar:: MINIO_AUDIT_KAFKA_QUEUE_SIZE
   :optional:

   .. include:: /includes/common-mc-admin-config.rst
      :start-after: start-minio-kafka-audit-logging-queue-size-desc
      :end-before: end-minio-kafka-audit-logging-queue-size-desc

   This environment variable corresponds with the :mc-conf:`audit_kafka.queue_size` configuration setting.

> Configuration Values

Reviewer 2: ```suggestion
Configuration Settings
```

--------------------

> These environment variables configure publishing regular :mc:`minio server` logs and audit logs to an HTTP webhook. 

Reviewer 2: guessing this was intended
```suggestion
These configuration settings configure publishing regular :mc:`minio server` logs and audit logs to an HTTP webhook. 
```

.. _minio-server-envvar-metrics-logging:

============================
Metrics and Logging Settings
============================

.. default-domain:: minio

.. contents:: Table of Contents
   :local:
   :depth: 2

This page covers settings that control behavior related to MinIO metrics and logging. 
See :ref:`minio-metrics-and-alerts` for more information.

These settings configure publishing regular :mc:`minio server` logs and audit logs to an HTTP webhook. 
See :ref:`minio-logging` for more complete documentation.

- :ref:`minio-server-envvar-logging-regular`
- :ref:`minio-server-envvar-logging-audit`
- :ref:`minio-server-envvar-logging-audit-kafka`

Prometheus Authentication
-------------------------

This setting controls how MinIO authenticates to Prometheus.

.. tab-set::

   .. tab-item:: Environment Variable
      :selected:

      .. envvar:: MINIO_PROMETHEUS_AUTH_TYPE

   .. tab-item:: Configuration Setting

>       There is not a configuration setting option.

Reviewer 2: ```suggestion
      This setting does not have a configuration variable setting.
```

Reviewer 2: Seems there a couple variants for the _nothing to see here_ text. I'm not sure which you prefer, I'll stop flagging them for the moment.

Author: I'll add an include for that to make sure they are consistent (and easier to change later, if need be). Then will update throughout.

The MinIO process must have read, write, and list access on the specified directory.

MinIO stores undelivered events in the specified store while the webhook service is offline and replays the stored events when connectivity resumes.
 
Queue Size
++++++++++

*Optional*

.. tab-set::

   .. tab-item:: Environment Variable
      :sync: envvar
      
      .. envvar:: MINIO_LOGGER_WEBHOOK_QUEUE_SIZE

   .. tab-item:: Configuration Setting
      :sync: config

      .. mc-conf:: logger_webhook queue_size
         :delimiter: " "

An integer value to use for the queue size for logger webhook targets.

.. _minio-server-envvar-logging-audit:
.. _minio-server-config-logging-audit:

Webhook Audit Logs
------------------

The following section documents environment variables for configuring MinIO to publish audit logs to an HTTP webhook endpoint. 
See :ref:`minio-logging-publish-audit-logs` for more complete documentation and tutorials on using these environment variables.

Multiple Targets
~~~~~~~~~~~~~~~~

You can specify multiple webhook endpoints as audit log targets by appending a unique identifier ``_ID`` for each set of related logging settings. 

> For example, the following command set two distinct audit log webhook endpoints:

Reviewer 2: likely a typo
```suggestion
For example, the following commands set two distinct audit log webhook endpoints:
```
